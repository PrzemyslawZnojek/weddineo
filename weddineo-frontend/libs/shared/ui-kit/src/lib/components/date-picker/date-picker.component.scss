:host {
  ::ng-deep {
    .mat-form-field {
      line-height: 1.3;
    }
    .mat-form-field-wrapper {
      padding-bottom: 1em;
    }
  }
}
