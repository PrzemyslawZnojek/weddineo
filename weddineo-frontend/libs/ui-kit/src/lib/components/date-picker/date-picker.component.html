<mat-form-field appearance="outline" [fxFlex]="width">
  <mat-label>{{label}}</mat-label>
  <input matInput [matDatepicker]="picker" [formControl]="fc" (focus)="picker.open()">
  <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
  <mat-datepicker #picker></mat-datepicker>
</mat-form-field>
