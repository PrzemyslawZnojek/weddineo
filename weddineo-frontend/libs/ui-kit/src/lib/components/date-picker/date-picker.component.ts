import { Component, Input } from '@angular/core';
import { FormControl } from '@angular/forms';

@Component({
  selector: 'weddi-date-picker',
  templateUrl: './date-picker.component.html',
  styleUrls: ['./date-picker.component.scss'],
})
export class DatePickerComponent {
  @Input() width: string = '100%';
  @Input() fc: FormControl;
  @Input() label: string;
}
