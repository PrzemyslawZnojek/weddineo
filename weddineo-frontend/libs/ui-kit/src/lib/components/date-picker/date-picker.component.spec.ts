import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FlexLayoutModule } from '@angular/flex-layout';
import { FormControl, ReactiveFormsModule } from '@angular/forms';
import { MatDatepickerModule } from '@angular/material/datepicker';
import { MatFormField, MatLabel } from '@angular/material/form-field';
import { MatInput } from '@angular/material/input';
import { MockComponents, MockModule } from 'ng-mocks';
import { DatePickerComponent } from './date-picker.component';

describe('DatePickerComponent', () => {
  let component: DatePickerComponent;
  let fixture: ComponentFixture<DatePickerComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        FlexLayoutModule,
        ReactiveFormsModule,
        MockModule(MatDatepickerModule),
      ],
      declarations: [
        DatePickerComponent,
        MockComponents(MatInput, MatFormField, MatLabel),
      ],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(DatePickerComponent);
    component = fixture.componentInstance;
    component.fc = new FormControl({});
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should render valid label', () => {
    component.label = 'test';
    fixture.detectChanges();
    const compiled = fixture.nativeElement;
    expect(compiled.querySelector('mat-label').innerHTML).toContain('test');
  });
});
