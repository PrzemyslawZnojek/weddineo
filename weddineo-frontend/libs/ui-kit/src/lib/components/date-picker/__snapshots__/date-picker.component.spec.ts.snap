// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerComponent should match snapshot 1`] = `
<weddi-date-picker
  fc={[Function FormControl]}
  width={[Function String]}
>
  <mat-form-field
    appearance="outline"
    style="box-sizing: border-box; flex: 1 1 100%; max-width: 100%;"
  >
    <mat-label>
      
    </mat-label>
    <input
      class="ng-untouched ng-pristine ng-valid"
      matinput=""
    />
    <mat-datepicker-toggle
      matsuffix=""
    />
    <mat-datepicker />
  </mat-form-field>
</weddi-date-picker>
`;
